sim/midas/src/main/cc/bridges/cpu_managed_stream.h

```diff
@@ -0,0 +1,107 @@
+// See LICENSE for license details.
+
+#ifndef __CPU_MANAGED_STREAM_H
+#define __CPU_MANAGED_STREAM_H
+
+#include <string>
+#include <functional>
```

Comment:
I guess my programming experience in functional languages is pushing me away from inheritance. I thought this might be a little easier to test, and it is a little bit less verbose to specialize for different hosts. I can change it though.

Reply:
No need to change.  I just wanted to capture the motivation somewhere in this PR for future reference.

Do you have tests for this stuff?